src/codeu/chat/server/Server.java

```diff
@@ -176,6 +177,19 @@ public void onMessage(InputStream in, OutputStream out) throws IOException {
       }
     });
 
+       // New Interest - A client wants to add a new interest to the back end.
+    this.commands.put(NetworkCode.NEW_INTERESTS_REQUEST,  new Command() {
+      @Override
+      public void onMessage(InputStream in, OutputStream out) throws IOException {
+
+        final Collection<Uuid> ids = Serializers.collection(Uuid.SERIALIZER).read(in);
```

Comment:
The way that I have it implemented now and the way that I think it should be implemented (just slack me if you want to talk about it) is that for each of these server methods, instead of sending a collection of Uuids, you just send the 2 discrete Uuids, the user's Uuid and the interest's Uuid.
So to add an interest, you need 2 pieces of information, the user's Uuid and the interest's Uuid. So this block would look like:
```
final Uuid userid = Uuid.SERIALIZER.read(in);
final Uuid interest = Uuid.SERIALIZER.read(in);
```